### PR TITLE
Bind `fetch` and `stream` in JS client

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

Bind `fetch` and `stream` in the JS client as these methods have no `this` value when called outside the client. 

Thanks to @pngwn for spotting the issue + solution!

Closes: #8707

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Review by pngwn — APPROVED** [review]
Looks great. Thanks for taking care of this @hannahblair!

**Comment by pngwn** [comment]
Hmm. The deployed demos work fine, not sure why those tests are failing.

**Comment by abidlabs** [comment]
Thanks @hannahblair! Can we add a test for this? I'm surprised that this wasn't already caught by our E2E tests

**Comment by pngwn** [comment]
Yeah I don't really understand how this didn't cause our functional tests to fail.

**Comment by pngwn** [comment]
Although now it is hilariously.

File: js/upload/src/UploadProgress.svelte [hunk 1]
```diff
@@ -59,6 +60,7 @@
 		};
 	});
 	onDestroy(() => {
+		// the stream will close itself but is here for perspicuity; remove .close() in 5.0
```
Comment (pngwn) [inline js/upload/src/UploadProgress.svelte:63]: the fuck kinda word is this

Comment (pngwn) [reply js/upload/src/UploadProgress.svelte:63]: Screenshot 2024-07-11 at 12 45 56

are ppl just making words up or something

Comment (pngwn) [reply js/upload/src/UploadProgress.svelte:63]: maybe i'm just stupider than i thought i was (and i already thought i was pretty stupid)

**Review by pngwn — APPROVED** [review]
Perfect, thanks! And thanks for figuring out that test failure @hannahblair!

**Pull request merged**